ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -358,10 +377,15 @@ private TransactionSelectionResult transactionSelectionResultForInvalidResult(
         LOG,
         "Delete invalid transaction {}, reason {}",
         transaction::toTraceLog,
-        invalidReasonValidationResult::getInvalidReason);
+        invalidReason::toString);
     return TransactionSelectionResult.DELETE_TRANSACTION_AND_CONTINUE;
   }
 
+  private boolean isTransientValidationError(final TransactionInvalidReason invalidReason) {
+    return invalidReason.equals(TransactionInvalidReason.GAS_PRICE_BELOW_CURRENT_BASE_FEE)
+        || invalidReason.equals(TransactionInvalidReason.INCORRECT_NONCE);
```

Comment:
Should we distinguish between nonce too low and nonce too high?  nonce too low will almost never become a valid transaction, especially if the nonce that used it's value is behind a finalized block.

Reply:
Good point, I have checked the `MainnetTransactionValidator` and found that INCORRECT_NONCE actually only means future nonce, since the check for nonce too low is done before and reported as NONCE_TOO_LOW, so nonce too low is not considered transient and will be removed, so we are aready good.

Reply:
can we rename incorrect nonce to NONCE_TOO_HIGH make it apparent on it's face that's what it's covering?  I want to remove the "secret knowledge" that incorrect_nonce only happens when nonce is to high.

Reply:
Make sense, and did the renaming, and since NONCE_TOO_HIGH was used when the nonce is above the max value, I had to renamve it to NONCE_OVERFLOW as well